js/ui/scheduler/ui.scheduler.appointment_model.js

```diff
@@ -492,10 +492,12 @@ class AppointmentModel {
     appointmentTakesSeveralDays(appointment) {
         const dataAccessors = this._dataAccessors;
         const startDate = dataAccessors.getter.startDate(appointment);
-        const endDate = dataAccessors.getter.endDate(appointment);
+        const endDate = new Date(dataAccessors.getter.endDate(appointment));
 
         const startDateCopy = dateUtils.trimTime(new Date(startDate));
-        const endDateCopy = dateUtils.trimTime(new Date(endDate));
+
+        // NOTE: subtract 1 millisecond to avoid 00.00 time
+        const endDateCopy = dateUtils.trimTime(new Date(endDate.getTime() - 1));
```

Comment:
To wrap in `new Date(...)` is unnecessary because of wrapping in `new Date(...)` is inside of `dateUtils.trimTime` function.